client/rpc/src/eth.rs

```diff
@@ -840,12 +899,31 @@ where
 	}
 
 	fn send_raw_transaction(&self, bytes: Bytes) -> BoxFuture<Result<H256>> {
-		let transaction = match rlp::decode::<ethereum::TransactionV0>(&bytes.0[..]) {
-			Ok(transaction) => transaction,
-			Err(_) => return Box::pin(future::err(internal_err("decode transaction failed"))),
+		let slice = &bytes.0[..];
+		if slice.len() == 0 {
+			return Box::pin(future::err(internal_err("transaction data is empty")));
+		}
+		let first = slice.get(0).unwrap();
+		let transaction = if first > &0x7f {
```

Comment:
verify this manually check still required after the recent changes in ethereum crate